internal/clusterinfo/data.go

```diff
@@ -0,0 +1,399 @@
+package clusterinfo
```

Comment:
all this simplification came out nice and now that we see the result it looks like there's clear opportunity to DRY it up in some future iteration (a lot of the methods looks suspiciously identical structurally)